From 52d6da6ce3406c00817ea762b11c4ab7929c1ace Mon Sep 17 00:00:00 2001
From: Simon Mikuda <simon.mikuda@streamunlimited.com>
Date: Thu, 6 Feb 2020 13:28:15 +0100
Subject: [PATCH] libfdt: do not use kernel-specific aliases
 [STREAMEIGHT-3041](thud)

Aliases without "__" for be/le conversions are not meant to be used outside
kernel code and different versions of kernel and u-boot expose or hide them
behind __KERNEL__, but it is cleaner to just use the full names. This works
around the include sequence mess in a yocto-thud build.

Signed-off-by: Andriy Danylovskyy <andriy.danylovskyy@streamunlimited.com>
---
 include/libfdt_env.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/libfdt_env.h b/include/libfdt_env.h
index 273b5d3..5630c4a 100644
--- a/include/libfdt_env.h
+++ b/include/libfdt_env.h
@@ -9,8 +9,8 @@
 #ifndef _LIBFDT_ENV_H
 #define _LIBFDT_ENV_H
 
-#include "compiler.h"
 #include "linux/types.h"
+#include <asm/byteorder.h>
 
 extern struct fdt_header *working_fdt;  /* Pointer to the working fdt */
 
@@ -18,10 +18,10 @@ typedef __be16 fdt16_t;
 typedef __be32 fdt32_t;
 typedef __be64 fdt64_t;
 
-#define fdt32_to_cpu(x)		be32_to_cpu(x)
-#define cpu_to_fdt32(x)		cpu_to_be32(x)
-#define fdt64_to_cpu(x)		be64_to_cpu(x)
-#define cpu_to_fdt64(x)		cpu_to_be64(x)
+#define fdt32_to_cpu(x)		__be32_to_cpu(x)
+#define cpu_to_fdt32(x)		__cpu_to_be32(x)
+#define fdt64_to_cpu(x)		__be64_to_cpu(x)
+#define cpu_to_fdt64(x)		__cpu_to_be64(x)
 
 /* adding a ramdisk needs 0x44 bytes in version 2008.10 */
 #define FDT_RAMDISK_OVERHEAD	0x80
-- 
2.7.4

